webhook_router/app.py

```diff
@@ -0,0 +1,47 @@
+#  Copyright 2024 Canonical Ltd.
+#  See LICENSE file for licensing details.
+
+"""Flask application which receives GitHub webhooks and logs those."""
+import json
+import logging
+
+from flask import Flask, request
+
+from webhook_router.validation import verify_signature
+
+WEBHOOK_SIGNATURE_HEADER = "X-Hub-Signature-256"
+
+app = Flask(__name__)
+app.config.from_prefixed_env()
+
+
+@app.route("/webhook", methods=["POST"])
+def handle_github_webhook() -> tuple[str, int]:
+    """Receive a GitHub webhook and append the payload to a file.
+
+    Returns:
+        A tuple containing an empty string and 200 status code.
+    """
+    if secret := app.config.get("WEBHOOK_SECRET"):
```

Comment:
I'm wondering if we could invert the if statements to reduce nesting
```suggestion
    if not secret := app.config.get("WEBHOOK_SECRET"):
        return "", 200
```

Reply:
L37-L39 have to be shared by both parts, so I would have to include them in your proposal, which seems to complicate the solution.